### PR TITLE
PYIC-4122: Added F2F CRI VC Message queue consumer side tests

### PR DESCRIPTION
**Pull request by RogVanGDS**
### What changed

Added F2F CRI VC Message queue consumer side tests

### Why did it change

F2F issues VCs asynchronously through a message queue, these tests were previously blocked, but are now ready.

### Issue tracking

- [PYIC-4122](https://govukverify.atlassian.net/browse/PYIC-4122)

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

[PYIC-4122]: https://govukverify.atlassian.net/browse/PYIC-4122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/JwtTestHelper.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package uk.gov.di.ipv.core.processasynccricredential.pact;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.util.Base64URL;
+
+public class JwtTestHelper {
+    private final String minifiedHeaderJson;
```
Comment (DanCorderIPV) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/JwtTestHelper.java]: Minor thing, but I just noticed that these could be base 64 encoded in the constructor too.

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 2]
```diff
@@ -0,0 +1,1348 @@
...
+import static au.com.dius.pact.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.when;
+import static uk.gov.di.ipv.core.processasynccricredential.helpers.AsyncCriResponseHelper.getAsyncResponseMessage;
+
+@Disabled("PACT tests should not be run in build pipelines at this time")
+@ExtendWith(PactConsumerTestExt.class)
+@ExtendWith(MockitoExtension.class)
+@PactTestFor(providerName = "PassportCriProvider")
```
Comment (DanCorderIPV) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: This isn't a passport test

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 3]
```diff
@@ -0,0 +1,1348 @@
...
+                .given("VC evidence verificationScore is 3")
+                .given("VC evidence strengthScore is 4")
+                .given("VC evidence type is IdentityCheck")
+                .given("VC evidence txn is eda339dd-aa83-495c-a4d4-75021e9415f9")
+                .given("VC jti is test-jti")
+                .expectsToReceive("A valid F2F CRI message from SQS")
+                .withContent(
+                        newJsonBody(
+                                        (body) -> {
+                                            final PactDslJsonBody actualPactDsl =
```
Comment (DanCorderIPV) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: I don't think this is needed?

**Review by DanCorderIPV — DISMISSED** [review]

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java (hunk 3 above)
Comment (DanCorderIPV) [reply lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: I don't think it's needed in any of the other pacts either :)

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 4]
```diff
@@ -0,0 +1,1323 @@
...
+                .given("VC evidence verificationScore is 3")
+                .given("VC evidence strengthScore is 4")
+                .given("VC evidence type is IdentityCheck")
+                .given("VC evidence txn is eda339dd-aa83-495c-a4d4-75021e9415f9")
+                .given("VC jti is test-jti")
+                .expectsToReceive("A valid F2F CRI with BRP Document")
+                .withContent(
+                        newJsonBody(
+                                        (body) -> {
+                                            final PactDslJsonBody actualPactDsl =
```
Comment (DanCorderIPV) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: Still one here

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 5]
```diff
@@ -0,0 +1,1317 @@
...
+                                "checkMethod": "vcrypt"
+                              },
+                              {
+                                "biometricVerificationProcessLevel": 3,
+                                "checkMethod": "bvr"
+                              }
+                            ],
+                            "validityScore": 2,
+                            "verificationScore": 3,
+                            "strengthScore": 4,
```
Comment (blakeyp) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: For a driving licence we'd expect a strength score of 3 (see https://github.com/govuk-one-login/architecture/blob/8018373649525bbe22e7a5bedafcf4b46c78e50e/rfc/0062-bank-account-verification.md?plain=1#L428)

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 6]
```diff
@@ -0,0 +1,1317 @@
...
```
Comment (blakeyp) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java:547]: Hmm I think it's validity 3 for BRP and I think the verification score might not be included (as it comes from the biometric verification score) - where did you get these from/could you confirm with the F2F team?

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java [hunk 7]
```diff
@@ -0,0 +1,1317 @@
...
+                              {
+                                "identityCheckPolicy": "published",
+                                "checkMethod": "vcrypt"
+                              },
+                              {
+                                "biometricVerificationProcessLevel": 3,
+                                "checkMethod": "bvr"
+                              }
+                            ],
+                            "validityScore": 2,
```
Comment (blakeyp) [inline lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java:478]: Also be good to get this confirmed

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java (hunk 5 above)
Comment (RogVanGDS) [reply lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: That link is to the BAV CRI, not the F2F one?

Comment (blakeyp) [reply lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java]: Wrong link sorry - https://github.com/govuk-one-login/architecture/blob/8018373649525bbe22e7a5bedafcf4b46c78e50e/rfc/0024-identity-vc-for-checks.md#expected-property-values

File: lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java (hunk 6 above)
Comment (RogVanGDS) [reply lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java:547]: They just pointed me towards the [documentation](https://govukverify.atlassian.net/wiki/spaces/FTFCRI/pages/3545792513/Generating+Verification+from+Yoti+Results), which states that strength, validation and verification are present. It doesn't make any exceptions for EEA Cards or BRPs

Comment (blakeyp) [reply lambdas/process-async-cri-credential/src/test/java/uk/gov/di/ipv/core/processasynccricredential/pact/f2fCri/ContractTest.java:547]: Ok I guess we'll confirm when the provider side gets implemented

**Comment by blakeyp** [comment]
✅ Looks good but there's a conflict

**Review by blakeyp — APPROVED** [review]

**Pull request merged**